lib/pbench/server/indexing_tarballs.py

```diff
@@ -116,26 +163,30 @@ def collect_tb(self, ARCHIVE_rp, qdir):
             raise
         except Exception:
             idxctx.logger.exception(
-                "Unexpected error encountered generating list"
-                " of tar balls to process"
+                "{} generating list of tar balls to process",
+                error_code["GENERIC_ERROR"].message,
             )
             # tuple to return the status and return value
-            return (12, [])
+            return (error_code["GENERIC_ERROR"].value, [])
         else:
             if not tarballs:
                 idxctx.logger.info("No tar balls found that need processing")
-                return (0, [])
 
-        return (0, sorted(tarballs))
+        return (error_code["OK"].value, sorted(tarballs))
+
+    def emit_error(self, logger_method, error, exception):
```

Comment:
There should really be a docstring here. Explain the purpose, the interpretation of `logger_method` (though the name change definitely helps), and it is probably worth pointing out that all log messages will originate from this frame rather than from the caller.

Reply:
What I had in mind was something a bit more complete. As an example:

```
"""Helper method to write a log message in a standard format from an error code

    Args
        logger_method -- Reference to a method of a Python logger object, like idxctx.logger.warning
        error -- An error code name from the Errors collection, like "OK"
        exception -- the original exception leading to the error

    Returns
        The numeric value of the error code

    Although all log messages will appear to have originated from this method, the origin can easily be
    identified from the error code value, and this interface provides simplicity and consistency.
"""

Reply:
I concur...but I'm always suspicious when the comment is longer than the code that it's documenting.